### PR TITLE
Allow to set traversal methods order

### PR DESCRIPTION
**Pull request by soffokl**
https://github.com/mysteriumnetwork/node/issues/3588

### CODE REVIEW

**Review by Snawoot — APPROVED** [review]

**Review by vkuznecovas — APPROVED** [review]

File: config/flags_network.go [hunk 1]
```diff
@@ -114,6 +114,12 @@ var (
 		Usage: "Range of UDP listen ports used for connections",
 		Value: "10000:60000",
 	}
+	// FlagTraversalOrder order of NAT traversal methods to be used for providing service.
+	FlagTraversal = cli.StringFlag{
+		Name:  "traversal",
+		Usage: "Comma separated order of NAT traversal methods to be used for providing service",
```
Comment (tomasmik) [inline config/flags_network.go:120]: Would it be useful to provide possible values here?

File: p2p/listener.go [hunk 2]
```diff
@@ -304,60 +293,23 @@ func (m *listener) providerStartConfigExchange(providerID identity.Identity, msg
 // required ports count for actual p2p and service connections and fallback to
 // acquiring extra ports for nat pinger if provider is behind nat, port mapping failed
 // and no manual port forwarding is enabled.
-func (m *listener) prepareLocalPorts(id, outboundIP string, tracer *trace.Tracer) (string, []int, []func(), error) {
+func (m *listener) prepareLocalPorts(id string, tracer *trace.Tracer) (string, []int, []func(), nat.StartPorts, error) {
```
Comment (tomasmik) [inline p2p/listener.go:296]: At some point the returns values here should be remade. Adding one more return here is just more confusion.

Comment (mdomasevicius) [reply p2p/listener.go:296]: It could be wrapped into struct with meaningful names attached

File: config/flags_network.go (hunk 1 above)
Comment (soffokl) [reply config/flags_network.go:120]: we are including all values as default. Here is how it looks in help:
```
   --traversal value                                      Comma separated order of NAT traversal methods to be used for providing service (default: "holepunching,upnp,manual")
```

File: p2p/nat/hole_punching.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func NewNatHolePunchingPortProvider() PortProvider {
+	udpPortRange, err := port.ParseRange(config.GetString(config.FlagUDPListenPorts))
+	if err != nil {
+		log.Warn().Err(err).Msg("Failed to parse UDP listen port range, using default value")
+	}
+
+	return &natHolePunchingPort{
+		pool:   port.NewFixedRangePool(*udpPortRange),
```
[the remark this answers is not in the document]
Comment (soffokl) [reply p2p/nat/hole_punching.go]: Yep, faced it with tests too, changed.

File: p2p/listener.go (hunk 2 above)
Comment (soffokl) [reply p2p/listener.go]: I'm going to change it with the next couple of PRs. So merging like this now.

**Pull request merged**